src/main/java/org/broadinstitute/consent/http/service/dao/NihServiceDAO.java

```diff
@@ -42,4 +42,17 @@ public void updateUserNihStatus(User user, NIHUserAccount nihAccount)
       userDAO.updateEraCommonsId(user.getUserId(), nihAccount.getNihUsername());
     });
   }
+
+  public void deleteNihAccountById(Integer userId) {
+    jdbi.useTransaction(handler -> {
+      UserDAO userDAO = handler.attach(UserDAO.class);
+      UserPropertyDAO userPropertyDAO = handler.attach(UserPropertyDAO.class);
+      Collection<UserProperty> properties = List.of(
+          new UserProperty(userId, UserFields.ERA_EXPIRATION_DATE.getValue()),
+          new UserProperty(userId, UserFields.ERA_STATUS.getValue())
+      );
+      userDAO.updateEraCommonsId(userId, null);
```

Comment:
This is a bug fix. Previously, we were not deleting this value when a user tried to delete their eRA Commons/RAS account link.